src/app/tags/components/TagsHeader/TagsHeader.tsx

```diff
@@ -1,46 +1,135 @@
 import { TagViewState } from "app/tags/types";
 
 export type Props = {
+  filter: TagSearchFilter;
+  setFilter: (filter: TagSearchFilter) => void;
+  searchText: string;
+  setSearchText: (searchText: string) => void;
+  isDetails: boolean;
   setSidePanelContent: SetSidePanelContent;
   tagViewState?: TagViewState | null;
+  onDelete: (id: Tag[TagMeta.PK], fromDetails?: boolean) => void;
 };
 
 export enum Label {
   CreateButton = "Create new tag",
+  EditButton = "Edit",
+  DeleteButton = "Delete",
   Header = "Tags header",
+  All = "All tags",
+  Manual = "Manual tags",
+  Auto = "Automatic tags",
 }
 
 export const TagsHeader = ({
+  filter,
+  setFilter,
+  searchText,
+  setSearchText,
+  isDetails,
   setSidePanelContent,
   tagViewState,
+  onDelete,
 }: Props): JSX.Element => {
-  const { machineCount } = useFetchMachineCount();
+  // Don't show the buttons when any of the forms are visible.
+  const showButtons = !tagViewState;
+  const id = useGetURLId(TagMeta.PK);
+  const tag = useSelector((state: RootState) =>
+    tagSelectors.getById(state, id)
+  );
+
   return (
-    <MachinesHeader
-      aria-label={Label.Header}
-      buttons={
-        tagViewState === TagViewState.Updating
-          ? null
-          : [
-              <Button
-                appearance="positive"
-                onClick={() =>
-                  setSidePanelContent({ view: TagSidePanelViews.AddTag })
-                }
-              >
-                {Label.CreateButton}
-              </Button>,
-            ]
-      }
-      machineCount={machineCount}
-      title="Tags"
-    />
+    <MainToolbar>
+      <MainToolbar.Title>Tags</MainToolbar.Title>
+      {isDetails ? (
+        <Link className="u-sv3" to={urls.tags.index}>
+          &lsaquo; Back to all tags
+        </Link>
+      ) : null}
+      <MainToolbar.Controls>
+        {tagViewState === TagViewState.Updating ? null : (
+          <>
+            {isDetails && tag ? (
+              <>
+                {showButtons ? (
+                  <>
+                    <Button
+                      element={Link}
+                      hasIcon
+                      state={{ canGoBack: true }}
+                      to={{
+                        pathname: urls.tags.tag.update({ id: tag.id }),
+                      }}
+                    >
+                      <Icon name="edit" /> <span>{Label.EditButton}</span>
+                    </Button>
+                    <Button
+                      appearance="negative"
+                      hasIcon
+                      onClick={() => onDelete(tag[TagMeta.PK], true)}
+                    >
+                      <Icon className="is-light" name="delete" />{" "}
+                      <span>{Label.DeleteButton}</span>
+                    </Button>
+                  </>
+                ) : null}
+              </>
+            ) : (
+              <>
+                <SearchBox
+                  className="u-no-margin--bottom"
```

Comment:
```suggestion
```

You will need to remove `.u-no-margin--bottom` from `.p-search-box`, bottom margins are controlled by the toolbar.

![Google Chrome screenshot 001320@2x](https://github.com/canonical/maas-ui/assets/7452681/1a8e9023-a044-4b96-b514-52eccda6843f)